Dynatrace/dynatrace-operator
# Remove the db from the provisioner logic (and codebase)

## Description
[DAQ-1412](https://dt-rnd.atlassian.net/browse/DAQ-1412)
[DAQ-1471](https://dt-rnd.atlassian.net/browse/DAQ-1471)

Removes the database code from the `provisioner` logic, and refactors it a bit, to hopefully make it more understandable.

Also removes the database code from the whole codebase:
- The `provisioner` was the last thing still using it
   - The linter complained after I removed the database code from there
     -   It was easier to remove the whole code than messing with the linter

## How can this be tested?

The provisioner refactor:
- It should work as normal, just no more database creation
- Look at code, see if it's clear

The database-code/CGO removal:
- Image still builds, but faster 😅 

[DAQ-1412]: https://dt-rnd.atlassian.net/browse/DAQ-1412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DAQ-1471]: https://dt-rnd.atlassian.net/browse/DAQ-1471?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
